tests/e2e/vsphere_shared_datastore.go

```diff
@@ -121,4 +126,158 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] "+
+		pv := persistentvolumes[0]
+		volHandle := pv.Spec.CSI.VolumeHandle
+		if guestCluster {
+			volHandle = getVolumeIDFromSupervisorCluster(volHandle)
+			gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+		}
+
+		// Create a Pod to use this PVC, and verify volume has been attached
+		ginkgo.By("Creating pod to attach PV to the node")
+		pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		var vmUUID string
+		var exists bool
+		ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", volHandle, pod.Spec.NodeName))
+		if vanillaCluster {
+			vmUUID = getNodeUUID(client, pod.Spec.NodeName)
+		} else if guestCluster {
+			vmUUID, err = getVMUUIDFromNodeName(pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		} else {
+			annotations := pod.Annotations
+			vmUUID, exists = annotations[vmUUIDLabel]
+			gomega.Expect(exists).To(gomega.BeTrue(), fmt.Sprintf("Pod doesn't have %s annotation", vmUUIDLabel))
+		}
+		framework.Logf("VMUUID : %s", vmUUID)
+
+		isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volHandle, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+
+		// Delete SC with Immediate Binding Mode
+		ginkgo.By("Delete SC created with Immediate Binding Mode")
+		client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Create SC with same name but with WaitForFirstConusmer Binding Mode
+		ginkgo.By("Recreate SC with same name but with WaitForFirstConusmer Binding Mode")
+		createStorageClass(client, scParameters, nil, "", storagev1.VolumeBindingWaitForFirstConsumer, false, storageclass.Name)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("storageclass name :%s", storageclass.GetName())
+
+		// Check Pod status after recreating Storage class with different binding mode
+		ginkgo.By("Verify Pod status after recreating Storage class with different binding mode")
+		pod_status := fpod.VerifyPodsRunning(client, namespace, pod.Name, true, 0)
+		if pod.Status.Phase == v1.PodRunning {
+			framework.Logf("Pod is in Running state after recreating Storage Class")
+
+		}
+		gomega.Expect(pod_status).NotTo(gomega.HaveOccurred())
+
+		// Check PVC status after recreating Storage class with different binding mode
+		ginkgo.By("Verify PVC status after recreating Storage class with different binding mode")
+		pvclaim, err = checkPvcHasGivenStatusCondition(client,
+			namespace, pvclaim.Name, false, "")
+		if v1.PersistentVolumePhase(pvclaim.Status.Phase) == v1.VolumeBound {
+			framework.Logf("PVC is in Bound Phase after recreating Storage Class")
+
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		// Deleting Pod
```

Comment:
Add delete pod in defer block after line 229

Reply:
acknowledged. Will update.